### PR TITLE
[kerbal] update to v2024.1.1

### PR DESCRIPTION
**Pull request by WentsingNee**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: versions/k-/kerbal.json [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "96765fcef80616b82190c2eeb1163ee856f96486",
+      "version": "2024.2.1",
+      "port-version": 0
+    },
+    {
+      "git-tree": "423247974576c87249311509a8d255d1435308f1",
+      "version": "2024.1.1",
+      "port-version": 0
+    },
```
Comment (JonLiu1993) [inline versions/k-/kerbal.json:12]:
```suggestion
```

Comment (JonLiu1993) [reply versions/k-/kerbal.json:12]: We only need to add "git-tree" once.

Comment (WentsingNee) [reply versions/k-/kerbal.json:12]: It's two steps of version update in one pull request (firstly update to v2024.1.1 and then update to v2024.2.1). Does it really require to change git tree only once?

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: versions/k-/kerbal.json (hunk 1 above)
Comment (dg0yt) [reply versions/k-/kerbal.json:12]: One PR can add only entry here.

Comment (WentsingNee) [reply versions/k-/kerbal.json:12]: you mean i just need to squash that two commits into one, right?

Comment (dg0yt) [reply versions/k-/kerbal.json:12]: No, you must actively remove the older version from this file. And that's the suggested change.
The state referenced by the older `git-tree` will never exist in the vcpkg repo.

Comment (WentsingNee) [reply versions/k-/kerbal.json:12]: but why some libs like `fmt` are allowed to keep the older version in git-tree?

versions/f-/fmt.json

```
{
  "versions": [
    {
      "git-tree": "74cfe848ded3d9610b11e442aa6b2341cfa65b1b",
      "version": "10.2.1",
      "port-version": 1
    },
    {
      "git-tree": "405156a2b01c91258bf66768ceb3ae75c1caba7f",
      "version": "10.2.1",
      "port-version": 0
    },
    {
      "git-tree": "f91bb86075f45f6fad0c35ab9e87d8ea43c9d389",
      "version": "10.2.0",
      "port-version": 0
    },
    {
      "git-tree": "dfe9aa860f5a8317f341a21d317be1cf44e89f18",
      "version": "10.1.1",
      "port-version": 0
    },
    {
      "git-tree": "d8209382ca192426861c6aa91d217ea8512bda46",
      "version": "10.1.0",
      "port-version": 0
    },
    {
      "git-tree": "eedb31bb1318118ec6a2d1bec60ab12e484092fd",
      "version": "10.0.0",
      "port-version": 0
    },
```

Comment (dg0yt) [reply versions/k-/kerbal.json:12]: We are not talking about  
*keeping older versions*  
but about  
*adding more than one version of a port in one PR*.

Comment (WentsingNee) [reply versions/k-/kerbal.json:12]: OK, I've got. Thank you for your explaination

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**